cumulus/client/bootnodes/src/advertisement.rs

```diff
@@ -207,9 +207,20 @@ impl BootnodeAdvertisement {
 	async fn handle_import_notification(&mut self, header: RelayHeader) {
```

Comment:
Some edge case that is not covered is when the notification stream skipped blocks (because of major sync). Then you would maybe advertise too long with an invalid key.

Reply:
Unfortunately, `stop_providing` is a local-only action only (Kademlia DHT doesn't have a way to tell target nodes we are not providing the key anymore), so remote nodes will advertise every key until it expires in 48 hours anyway.

Reply:
This was not what I wanted to say :) The problem I raised there is when you for example sync across an epoch change, but the node was doing this in "major sync mode". Then you would not get a notification for these blocks.

Reply:
Thanks for the explanation, this makes sense. How common is such situation except during the initial sync?